### PR TITLE
[test] undo truncation of digits in btc amount

### PR DESCRIPTION
**Pull request by ghost**
- Undo truncation of decimal places of btc amount when calculating fee_expected as per this comment: [#19940 (comment) ](https://github.com/bitcoin/bitcoin/pull/19940/files#r491314005)

- Converted fee_expected to Decimal as per this comment: [#19940 (comment) ](https://github.com/bitcoin/bitcoin/pull/19940/files#r491314844)

### CODE REVIEW

File: test/functional/mempool_accept.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def run_test(self):
             locktime=node.getblockcount() + 2000,  # Can be anything
         ))['hex']
         tx.deserialize(BytesIO(hex_str_to_bytes(raw_tx_final)))
-        fee_expected = int(coin['amount']) - output_amount
+        fee_expected = Decimal(str(coin['amount']) )- Decimal(str(output_amount))
```
Comment (kallewoof) [inline test/functional/mempool_accept.py:108]: Odd spacing. Try to adapt the style around you (i.e. spaces before and after `-`, no space between `)` and `)`, ...

**Comment by fanquake** [comment]
Thanks for the contribution, however it looks like this is already being taken care of as part of #20039 (which is more extensive).

File: test/functional/mempool_accept.py (hunk 1 above)
Comment (riyasingh0799) [reply test/functional/mempool_accept.py]: Thank you @kallewoof. I'll be careful about styling from now on.

**Comment by fanquake** [comment]
Closing this in favour of #20039. @riyasingh0799 maybe you'd like to help review that PR instead?

**Pull request closed without merging**